memgpt/connectors/db.py

```diff
@@ -1,329 +1,432 @@
+
+            __abstract__ = True  # this line is necessary
+
+            # Assuming passage_id is the primary key
+            # id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+            id = Column(CommonUUID, primary_key=True, default=uuid.uuid4)
+            # id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
+            user_id = Column(String, nullable=False)
+            text = Column(String, nullable=False)
+            doc_id = Column(String)
+            agent_id = Column(String)
+            data_source = Column(String)  # agent_name if agent, data_source name if from data source
+
+            # vector storage
+            if dialect == "sqlite":
+                print("DIALECT", dialect, "using BLOB")
+                embedding = Column(CommonVector)
+            else:
+                from pgvector.sqlalchemy import Vector
+
+                print("DIALECT", dialect, "using VECTOR")
+                embedding = mapped_column(Vector(config.embedding_dim))
+
+            metadata_ = Column(MutableJson)
+
+            def __repr__(self):
+                return f"<Passage(passage_id='{self.id}', text='{self.text}', embedding='{self.embedding})>"
+
+            def to_record(self):
+                return Passage(
+                    text=self.text,
+                    embedding=self.embedding,
+                    doc_id=self.doc_id,
+                    user_id=self.user_id,
+                    id=self.id,
+                    data_source=self.data_source,
+                    agent_id=self.agent_id,
+                    metadata=self.metadata_,
+                )
+
+        """Create database model for table_name"""
+        class_name = f"{table_name.capitalize()}Model" + dialect
+        return create_or_get_model(class_name, PassageModel, table_name)
+
+    elif table_type == TableType.RECALL_MEMORY:
+
+        class MessageModel(Base):
+            """Defines data model for storing Message objects"""
+
+            __abstract__ = True  # this line is necessary
+
+            # Assuming message_id is the primary key
+            # id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+            id = Column(CommonUUID, primary_key=True, default=uuid.uuid4)
+            # id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
+            user_id = Column(String, nullable=False)
+            agent_id = Column(String, nullable=False)
+
+            # openai info
+            role = Column(String, nullable=False)
+            text = Column(String)  # optional: can be null if function call
+            model = Column(String, nullable=False)
+            user = Column(String)  # optional: multi-agent only
+
+            # tool call request info
+            # if role == "assistant", this MAY be specified
+            # if role != "assistant", this must be null
+            # TODO align with OpenAI spec of multiple tool calls
```

Comment:
@sarahwooders I think we want to align this more with OAI's spec:

<img width="569" alt="image" src="https://github.com/cpacker/MemGPT/assets/5475622/0c7b3d00-b22f-4d4a-9ccc-cde0e8617fcc">

Reply:
Leaving to #783 in future merge / PR